src/gist.rs

```diff
@@ -109,6 +114,28 @@ impl GistClient {
         Self::get_response(response)
     }
 
+    fn gist_id_from_url<'a>(&self, gist_url: &'a str) -> Option<&'a str> {
+        let re = Regex::new(
+            // Expect URL like https://gist.github.com/<user>/<gist_id>
+            r"https://gist\.github\.com/.+/(?P<gist_id>[0-9a-f]+)$",
+        )
+        .unwrap();
+        re.captures(gist_url)
+            .and_then(|cap| cap.name("gist_id").map(|gist_id| gist_id.as_str()))
+    }
+
+    /// Retrieve a Gist by URL
+    pub fn get_gist_by_url(&self, gist_url: &str) -> color_eyre::Result<Gist> {
+        let gist_id = self.gist_id_from_url(gist_url);
+        match gist_id {
+            Some(gist_id) => self.get_gist(gist_id),
```

Comment:
If `get_gist` throws an error here, would be nice to add a suggestion - "Make sure this is a valid Gist ID and the Gist is public. If it's private and you have access, set $THE_WAY_GITHUB_TOKEN" (or something like that)

Reply:
I found out that you don't actually need access token even for private Gists (aka secret Gists).
Here is secret gist I made, you should be able to access it without being logged in (e.g. via your browser's incognito mode):
https://gist.github.com/xiaochuanyu/e5deab8d78ce838f22f160c9b14daf17
The only secret part is the ID.

Reply:
Oh, learnt something new!